packages/utils/src/abortableTimers.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * setTimeout with AbortSignal support. Aborting will simply clear
+ * the timeout silently.
+ */
+export const setAbortableTimeout = createAbortableTimerFn(setTimeout, clearTimeout, true)
+
+/**
+ * setInterval with AbortSignal support. Aborting will simply clear
+ * the interval silently.
+ */
+export const setAbortableInterval = createAbortableTimerFn(setInterval, clearInterval, false)
+
+function createAbortableTimerFn(
+    setupTimerFn: (cb: () => void, ms?: number) => NodeJS.Timer,
+    clearFn: (ref: NodeJS.Timer) => void,
+    removeListenerOnCb: boolean
+): (cb: () => void, ms?: number, abortSignal?: AbortSignal) => NodeJS.Timer {
```

Comment:
We need to implement a custom cleanup to make sure that abort listeners really get cleared otherwise we risk memory leak issues. We should change this method not to return a `NodeJS.Timer` but a `{ clear(): void }`.

Reply:
Fixed in 1299520 and 75908a3